compiler.cpp

```diff
@@ -464,40 +423,65 @@ void SetType(map<ASTNode*, MiniAPLArrayType>& Types, ASTNode* Expr) {
   } else if (Expr->GetType() == EXPR_TYPE_SCALAR) {
     Types[Expr] = {{1}};
   } else if (Expr->GetType() == EXPR_TYPE_VARIABLE) {
-    string ExprName = static_cast<VariableASTNode*>(Expr)->Name;
+    std::string ExprName = static_cast<VariableASTNode *>(Expr)->Name;
     for (auto T : Types) {
       auto V = T.first;
       if (V->GetType() == EXPR_TYPE_VARIABLE) {
-        string Name = static_cast<VariableASTNode*>(V)->Name;
+        std::string Name = static_cast<VariableASTNode *>(V)->Name;
         if (Name == ExprName) {
           Types[Expr] = T.second;
         }
       }
     }
   }
+}
 
+// Run a set of optimization passes on this module.
```

Comment:
Do we need optimization passes? most programs will constant fold for this assignment